localstack/services/dynamodb/resource_providers/aws_dynamodb_globaltable.py

```diff
@@ -0,0 +1,392 @@
+            request.custom_context[REPEATED_INVOCATION] = True
+
+            if not model.get("TableName"):
+                model["TableName"] = util.generate_default_name(
+                    stack_name=request.stack_name, logical_resource_id=request.logical_resource_id
+                )
+
+            create_params = util.select_attributes(
+                model,
+                [
+                    "AttributeDefinitions",
+                    "BillingMode",
+                    "GlobalSecondaryIndexes",
+                    "KeySchema",
+                    "LocalSecondaryIndexes",
+                    "Replicas",
+                    "SSESpecification",
+                    "StreamSpecification",
+                    "TableName",
+                    "TimeToLiveSpecification",
+                    "WriteProvisionedThroughputSettings",
+                ],
+            )
+
+            replicas = create_params.pop("Replicas", [])
+
+            creation_response = request.aws_client_factory.dynamodb.create_table(**create_params)
+            model["Arn"] = creation_response["TableDescription"]["TableArn"]
+            model["TableId"] = creation_response["TableDescription"]["TableId"]
+
+            if creation_response["TableDescription"].get("LatestStreamArn"):
+                model["StreamArn"] = creation_response["TableDescription"]["LatestStreamArn"]
+
+            replicas_to_create = []
+            for replica in replicas:
+                create = {
+                    "RegionName": replica.get("Region"),
+                    "KMSMasterKeyId": replica.get("KMSMasterKeyId"),
+                    "ProvisionedThroughputOverride": replica.get("ProvisionedThroughputOverride"),
+                    "GlobalSecondaryIndexes": replica.get("GlobalSecondaryIndexes"),
+                    "TableClassOverride": replica.get("TableClassOverride"),
+                }
+
+                create = {k: v for k, v in create.items() if v is not None}
+
+                replicas_to_create.append({"Create": create})
+
+                request.aws_client_factory.dynamodb.update_table(
+                    ReplicaUpdates=replicas_to_create, TableName=model["TableName"]
+                )
+
+            return ProgressEvent(
+                status=OperationStatus.IN_PROGRESS,
+                resource_model=model,
+                custom_context=request.custom_context,
+            )
+
+        status = request.aws_client_factory.dynamodb.describe_table(TableName=model["TableName"])[
+            "Table"
+        ]["TableStatus"]
+        if status == "ACTIVE":
+            return ProgressEvent(
+                status=OperationStatus.SUCCESS,
+                resource_model=model,
+                custom_context=request.custom_context,
+            )
+
+        return ProgressEvent(
+            status=OperationStatus.IN_PROGRESS,
+            resource_model=model,
+            custom_context=request.custom_context,
+        )
```

Comment:
too generic, only `status == "CREATING"` (and maybe `UPDATING`) should result in a returned `IN_PROGRESS` state.